vcd/metadata_openapi_common_test.go

```diff
@@ -0,0 +1,282 @@
+//go:build rde || functional || ALL
+
+package vcd
+
+import (
+	"fmt"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+	"github.com/vmware/go-vcloud-director/v2/types/v56"
+	"strconv"
+	"strings"
+	"testing"
+)
+
+// testOpenApiMetadataEntryCRUD executes a test that asserts CRUD operation behaviours of "metadata_entry" attribute in the given HCL
+// templates, that must correspond to a resource and a data source referencing this resource.
+// The HCL template requires {{.Name}} and {{.Metadata}} fields, and the usual {{.Org}} and {{.Vdc}}.
+// You can add extra parameters as well to inject in the given HCL template, or override these mentioned ones.
+// The data source HCL is always concatenated to the resource after creation, and it's skipped on binary tests.
+func testOpenApiMetadataEntryCRUD(t *testing.T, resourceTemplate, resourceAddress, datasourceTemplate, datasourceAddress string, extraParams StringMap) {
+	preTestChecks(t)
+	var params = StringMap{
+		"Org":  testConfig.VCD.Org,
+		"Vdc":  testConfig.Nsxt.Vdc,
+		"Name": t.Name(),
+	}
+
+	// This output allows to perform some assertions on the ID inside a TypeSet,
+	// which is impossible to obtain otherwise.
+	outputHcl := `
```

Comment:
Any chance we can improve the formatting of this generated code?
Here's an example:
```
resource "vcd_rde" "test_rde" {
  org          = "System"
  rde_type_id  = data.vcd_rde_type.rde_type.id
  name         = "TestAccVcdRdeMetadata"
  input_entity = "{\"foo\":\"bar\"}" # We are just testing metadata so we don't care about entity state
  resolve      = true

		  metadata_entry {
			key         = "stringKey1"
			value       = "stringValue1"
			type        = "StringEntry"
			domain      = "TENANT"

			readonly    = false
            persistent  = false
		  }
		  metadata_entry {
			key         = "numberKey1"
			value       = "1"
			type        = "NumberEntry"
			domain      = "TENANT"

			readonly    = false
            persistent  = false
		  }
		  metadata_entry {
			key         = "boolKey1"
			value       = "false"
			type        = "BoolEntry"
			domain      = "TENANT"

			readonly    = false
            persistent  = false
		  }
		  metadata_entry {
			key         = "readOnly1"
			value       = "readOnly1"
			type        = "StringEntry"
			domain      = "TENANT"

			readonly    = true
            persistent  = false
		  }
		  metadata_entry {
			key         = "namespace"
			value       = "namespace1"
			type        = "StringEntry"
			domain      = "TENANT"
            namespace   = "namespace1"
			readonly    = false
            persistent  = false
		  }
		  metadata_entry {
			key         = "namespace"
			value       = "namespace2"
			type        = "StringEntry"
			domain      = "TENANT"
            namespace   = "namespace2"
			readonly    = false
            persistent  = false
		  }
		  metadata_entry {
			key         = "provider1"
			value       = "provider1"
			type        = "StringEntry"
			domain      = "PROVIDER"

			readonly    = false
            persistent  = false
		  }
		  metadata_entry {
			key         = "persistent1"
			value       = "persistent1"
			type        = "StringEntry"
			domain      = "TENANT"

			readonly    = false
            persistent  = true
		  }
}

		output "metadata_id" {
           value = tolist(vcd_rde.test_rde.metadata_entry)[0].id
        }
```
It doesn't have to be perfect, but a bit more readable would be good

Reply:
Should be fixed now. I also amended identation in the XML metadata tests